extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/validator/LDAPCredentialsValidator.java

```diff
@@ -235,6 +245,7 @@ boolean validatePassword(BasicAuthLDAPConfig ldapConfig, LdapName userDn, char[]
         LOG.warn("Exception closing LDAP context");
         // ignored
       }
+      Thread.currentThread().setContextClassLoader(currentClassLoader);
```

Comment:
Should the classLoader be reset in a `finally` block? 
```suggestion
  finally {
      Thread.currentThread().setContextClassLoader(currentClassLoader);
  }
```